.github/workflows/test-all.yml

```
@@ -0,0 +1,678 @@
+            echo "Setting release type to dry-run"
+            echo "RELEASE_TYPE=dry-run" >> $GITHUB_OUTPUT
+          fi
+  prepare_package_for_release:
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - name: Use Node.js 18
+        uses: actions/setup-node@v4.0.0
+        with:
+          node-version: 18
+          cache: 'yarn'
+      - name: Install dependencies
+        run: yarn install --non-interactive
+      - name: "Set new react-native version and commit changes"
+        env:
+          CIRCLE_BRANCH: ${{ github.ref_name }}
+        run: |
+          LATEST=false
+          DRYRUN=true
+          VERSION=$(grep '"version"' package.json | cut -d '"' -f 4 | head -1)
+          echo "Using the version from the package.json: $VERSION"
+          node ./scripts/releases-ci/prepare-package-for-release.js -v "$VERSION" -l $LATEST --dry-run $DRYRUN
+  prepare_hermes_workspace:
+    runs-on: ubuntu-latest
+    env:
+      HERMES_WS_DIR: /tmp/hermes
+      HERMES_VERSION_FILE: packages/react-native/sdks/.hermesversion
+      BUILD_FROM_SOURCE: true
+    outputs:
+      react-native-version: ${{ steps.react-native-version.outputs.version }}
+      hermes-version: ${{ steps.hermes-version.outputs.version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+      - name: Use Node.js 18
+        uses: actions/setup-node@v4.0.0
+        with:
+          node-version: 18
+          cache: 'yarn'
+      - name: Setup hermes version
+        id: hermes-version
+        run: |
+          mkdir -p "/tmp/hermes" "/tmp/hermes/download" "/tmp/hermes/hermes"
+
+          if [ -f "$HERMES_VERSION_FILE" ]; then
+            echo "Hermes Version file found! Using this version for the build:"
+            echo "VERSION=$(cat $HERMES_VERSION_FILE)" >> "$GITHUB_OUTPUT"
+          else
+            echo "Hermes Version file not found!!!"
+            echo "Using the last commit from main for the build:"
+            HERMES_TAG_SHA=$(git ls-remote https://github.com/$GITHUB_REPOSITORY main | cut -f 1 | tr -d '[:space:]')
+            echo "VERSION=$HERMES_TAG_SHA" >> "$GITHUB_OUTPUT"
+          fi
+          echo "Hermes commit is $HERMES_TAG_SHA"
+      - name: Get react-native version
+        id: react-native-version
+        run: |
+          VERSION=$(cat packages/react-native/package.json | jq -r '.version')
+          # Save the react native version we are building in an output variable so we can use that file as part of the cache key.
+          echo "VERSION=$VERSION" >> "$GITHUB_OUTPUT"
+          echo "React Native Version is $VERSION"
+      - name: Cache hermes workspace
+        uses: actions/cache@v4.0.0
+        with:
+          path: |
+            /tmp/hermes/download/
+            /tmp/hermes/hermes/
+          key: v5-hermes-${{ github.run_number }}-${{ steps.hermes-version.outputs.version }}
+          enableCrossOsArchive: true
+      - name: Yarn- Install Dependencies
```

nit: Align with other steps named `Install dependencies`, omitting Yarn detail.

```suggestion
      - name: Install dependencies
```